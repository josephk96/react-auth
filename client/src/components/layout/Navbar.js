import React, { useContext } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import breakpoint, { map } from "styled-components-breakpoint";

import AuthContext from "../../context/auth/authContext";

const NavContainer = styled.div`
top: 0;
position: sticky;
  width: 100%;
  grid-area: navbar;
`;

const Nav = styled.nav`
  background-color: #60adf0;
  height: 100%;
  display: flex;
  justify-content: center;
  ${breakpoint("md")`
    justify-content: flex-end;
    `}
`;

const LoggedInNav = styled.nav`
  height: 100%;
  background-color: #60adf0;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  ${breakpoint("md")`
    justify-content: flex-end;
    `}
`;

const List = styled.ul`
  display: flex;
  align-items: center;
  font-size: 1.5rem;
  justify-content: center;
  font-family: "Lato", sans-serif;
  ${breakpoint("md")`
    justify-content: flex-end;
    margin-top: 1.5rem;
    `}
`;

const StyledLink = styled(Link)`
  text-decoration: none;
  color: white;
  padding: 1.5rem;
  ${breakpoint("md")`
    height: 3.5rem;
    margin-right: 3rem;
    `}
`;

const StyledLogout = styled(Link)`
  text-decoration: none;
  font-size: 3rem;
  color: white;
  padding: 1.5rem;
  justify-self: flex-start;

  ${breakpoint("md")`
    height: 3.5rem;
    margin-right: 3rem;
    `}
`;

StyledLogout.displayName = 'StyledLogout';

const Navbar = () => {
  const authContext = useContext(AuthContext);
  const { isAuthenticated, logout } = authContext;

  return (
    <>
      {isAuthenticated ? (
        <NavContainer>
          <LoggedInNav>
            <List>
              <StyledLogout onClick={() => logout()}>
                <i class="fas fa-sign-out-alt"></i>
              </StyledLogout>
            </List>
          </LoggedInNav>
        </NavContainer>
      ) : (
        <NavContainer>
          <Nav>
            <List>
              <StyledLink to="/login">Login</StyledLink>
              <StyledLink to="/register">Register</StyledLink>
            </List>
          </Nav>
        </NavContainer>
      )}
    </>
  );
}

export default Navbar
