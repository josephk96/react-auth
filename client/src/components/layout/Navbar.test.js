import React from "react";
import Navbar from "./Navbar";
import { shallow } from "enzyme";

jest.mock("react", () => {
  const ActualReact = require.requireActual("react");
  return {
    ...ActualReact,
    useContext: () => ({
      authContext: {}
    })
  };
});

describe("Navbar", () => {
  it("renders without crashing", () => {
    const component = shallow(<Navbar />);
    expect(component.debug()).toMatchSnapshot();
  });
});
