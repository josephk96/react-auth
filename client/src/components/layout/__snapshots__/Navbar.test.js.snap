// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders without crashing 1`] = `
"<Fragment>
  <styled.div>
    <styled.nav>
      <styled.ul>
        <Styled(Link) to=\\"/login\\">
          Login
        </Styled(Link)>
        <Styled(Link) to=\\"/register\\">
          Register
        </Styled(Link)>
      </styled.ul>
    </styled.nav>
  </styled.div>
</Fragment>"
`;
